src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -26,16 +26,25 @@ object WdlExpression {
 
   val unaryOperators = Set("LogicalNot", "UnaryPlus", "UnaryNegation")
 
-  def evaluate(ast: AstNode, lookup: ScopedLookupFunction, functions: WdlFunctions): Try[WdlValue] = {
+  def interpolate(str: String, lookup: ScopedLookupFunction): String =
+    "\\$\\{([a-zA-Z0-9_]+)\\}".r.findAllIn(str).foldLeft(str) { (string, tag) =>
```

Comment:
The `findAllIn` API returns a `MatchIterator` which would appear to make the complexity of a fold unnecessary.  Pulling out the logic to a HOF as suggested below still seems like a good idea.  :smile: